src/components/calculator.js

```diff
@@ -0,0 +1,225 @@
+import React, { Component } from 'react';
+import calculate from '../logic/calculate';
+
+class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  onClickButton = (e) => {
+    const newState = calculate(this.state, e.target.name);
+    this.setState(newState);
+  };
+
+  render() {
+    const { total } = this.state;
+    let result = total === null ? 0 : total;
+    const { next } = this.state;
+    result = next !== null ? next : result;
+
+    return (
+      <div id="calculator">
+        <input
+          id="result"
+          type="text"
+          name="calculator-screen"
+          value={result}
+        />
+        <div id="calcButtons">
+          <div>
+            <button
+              onClick={this.onClickButton}
+              name="AC"
+              type="button"
+              className="calcButton"
+              id="clear"
+            >
+              AC
+            </button>
```

Comment:
- [ ] You did a great job developing `Calculator.js` :+1:, but as per the project requirements, you need to create at least one child component for `Calculator.js`.